purchases-capacitor-ui/ios/Plugin/Plugin/RevenueCatUIPlugin.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import Capacitor
+import PurchasesHybridCommonUI
+
+/**
+ * RevenueCat UI Plugin for Capacitor
+ * Based on the official RevenueCat Flutter UI SDK approach
+ */
+@objc(RevenueCatUIPlugin)
+public class RevenueCatUIPlugin: CAPPlugin {
+
+    // MARK: - Properties
+
+    private var savedCall: CAPPluginCall?
+
+    /// PaywallProxy for iOS 15.0+, will be nil on older iOS versions
+    private var _paywallProxy: PaywallProxyType?
+
+    /// CustomerCenterProxy for iOS 15.0+, will be nil on older iOS versions
+    private var _customerCenterProxy: CustomerCenterProxyType?
+
+    // MARK: - Initialization
+
+    override init() {
+        super.init()
+
+        // Initialize proxies if iOS 15.0+ is available
+        if #available(iOS 15.0, *) {
+            _paywallProxy = PaywallProxy()
+            _customerCenterProxy = CustomerCenterProxy()
+        }
+    }
+
+    // MARK: - Plugin Methods
+
+    @objc func presentPaywall(_ call: CAPPluginCall) {
+        // Ensure UI operations run on the main thread
+        DispatchQueue.main.async { [weak self] in
```

Comment:
Wouldn't it be better to make `RevenueCatUIPlugin` `@MainActor` ? Or is there a limitation with capacitor? Just asking ...